src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/commonmodels/Address.java

```diff
@@ -155,16 +155,26 @@ public boolean equals(Object o) {
                 && StringUtils.equalsIgnoreCase(normalise(poBox), normalise(address.poBox))
                 && StringUtils.equalsIgnoreCase(normalise(careOfCompany),
                 normalise(address.careOfCompany))
-                && StringUtils.equalsIgnoreCase(normalise(houseNameNum),
-                normalise(address.houseNameNum))
-                && StringUtils.equalsIgnoreCase(normalise(street), normalise(address.street))
+                && StringUtils.equalsIgnoreCase(
+                        normalise(getPremisesAndStreet()),
+                        normalise(address.getPremisesAndStreet()))
                 && StringUtils.equalsIgnoreCase(normalise(area), normalise(address.area))
                 && StringUtils.equalsIgnoreCase(normalise(postTown), normalise(address.postTown))
                 && StringUtils.equalsIgnoreCase(normalise(region), normalise(address.region))
                 && StringUtils.equalsIgnoreCase(normalise(postCode), normalise(address.postCode))
                 && StringUtils.equalsIgnoreCase(normalise(country), normalise(address.country));
     }
 
+    private String getPremisesAndStreet() {
+        if (StringUtils.isEmpty(houseNameNum)) {
```

Comment:
Could do this using StringBuilder class

```suggestion
       StringBuilder sb = new StringBuilder();
       sb.append(houseNameNum);
       sb.append(" " + street);
       return sb.toString().trim();
```

Reply:
Not a big issue- just seen that the code has been merged already so all good! :)

Reply:
StringJoiner could have been used but as it's only two items. We need to follow this up with some re-work as we have address comparisons in two places.